tests/main/apparmor-prompting-flag-restart/task.yaml

```diff
@@ -0,0 +1,100 @@
+    # Check that /v2/system-info reports prompting disabled
+    curl -sS --unix-socket /run/snapd.socket http://localhost/v2/system-info | jq -r '.result.features."apparmor-prompting".enabled' | MATCH "false"
+
+    check_restart_and_setting() {
+        # Wait a second to give snapd a chance to restart
+        sleep 1
+        # Check that snapd restarted
+        systemctl is-active snapd.service snapd.socket
+        NEW_SNAPD_PID="$(systemctl show --property MainPID snapd.service | cut -f2 -d=)"
+        test ! "$NEW_SNAPD_PID" = "$SNAPD_PID"
+        # Set SNAPD_PID so future checks use most recent PID
+        SNAPD_PID="$NEW_SNAPD_PID"
+        # Check that /v2/snaps/system/conf reports prompting flag set correctly
+        curl -sS --unix-socket /run/snapd.socket http://localhost/v2/snaps/system/conf | jq -r '.result.experimental."apparmor-prompting"' | MATCH "$1"
+        # Check that /v2/system-info reports prompting correctly
+        curl -sS --unix-socket /run/snapd.socket http://localhost/v2/system-info | jq -r '.result.features."apparmor-prompting".enabled' | MATCH "$1"
+    }
+
+    # Enable prompting via snap client
+    snap set system experimental.apparmor-prompting=true
+
+    echo "Check that snapd restarted after prompting set to true via snap client"
+    check_restart_and_setting "true"
+
+    # Disable prompting via snap client
+    snap set system experimental.apparmor-prompting=false
+
+    echo "Check that snapd restarted after prompting set to false via snap client"
+    check_restart_and_setting "false"
+
+    # Enable prompting via API request
+    curl -sS --unix-socket /run/snapd.socket http://localhost/v2/snaps/system/conf -X PUT -d '{"experimental.apparmor-prompting": true}' | jq -r '.status' | MATCH "Accepted"
+
+    echo "Check that snapd restarted after prompting set to true via curl"
+    check_restart_and_setting "true"
+
+    # Disable prompting via API request
+    curl -sS --unix-socket /run/snapd.socket http://localhost/v2/snaps/system/conf -X PUT -d '{"experimental.apparmor-prompting": false}' | jq -r '.status' | MATCH "Accepted"
+
+    echo "Check that snapd restarted after prompting set to false via curl"
+    check_restart_and_setting "false"
+
+    # Check that setting the same value multiple times does not restart snapd
+    for value in true false; do
+        # Initially set value, which will trigger a restart
+        snap set system experimental.apparmor-prompting="$value"
+        # Wait a second to give snapd the chance to restart
+        sleep 1
+        systemctl is-active snapd.service snapd.socket
+        # Get PID after initial value set
+        SNAPD_PID="$(systemctl show --property MainPID snapd.service | cut -f2 -d=)"
+
+        # Set same value again
+        snap set system experimental.apparmor-prompting="$value"
+        # Wait a second to let snapd handle the change
+        sleep 1
+        systemctl is-active snapd.service snapd.socket
```

Comment:
you should probably retry this check instead until is true, same in other places doing this. I'm also not 100% sure if is-active is the right check or not

Reply:
If it's not active, it means snapd has restarted, which should throw an error, since restarts shouldn't occur when setting the flag to the same value again. But `snap changes` might be a better check here: If snapd restarts, then `snap changes` will return `error: daemon is restarting`, otherwise it will block until the change finishes and then return a new change entry:

```
root@lenient-sheep:~# snap set system experimental.apparmor-prompting=false
root@lenient-sheep:~# snap changes
ID   Status  Spawn               Ready               Summary
25   Done    today at 21:58 UTC  today at 21:58 UTC  Change configuration of "core" snap
26   Done    today at 21:59 UTC  today at 21:59 UTC  Change configuration of "core" snap
27   Done    today at 21:59 UTC  today at 21:59 UTC  Change configuration of "core" snap

root@lenient-sheep:~# snap set system experimental.apparmor-prompting=true; snap changes
error: daemon is restarting
root@lenient-sheep:~# snap set system experimental.apparmor-prompting=true; snap changes
ID   Status  Spawn               Ready               Summary
25   Done    today at 21:58 UTC  today at 21:58 UTC  Change configuration of "core" snap
26   Done    today at 21:59 UTC  today at 21:59 UTC  Change configuration of "core" snap
27   Done    today at 21:59 UTC  today at 21:59 UTC  Change configuration of "core" snap
28   Done    today at 22:00 UTC  today at 22:00 UTC  Change configuration of "core" snap
29   Done    today at 22:00 UTC  today at 22:00 UTC  Change configuration of "core" snap
```

If we want to be very careful, we could call `snap changes` before changing the value and then compare it with the result after and confirm it's the same, but I think that's unnecessary given that state seems to be locked until the change finishes, so the result is always a new change entry with status `Done`.